src/Test/Perf/infra/fetch_build.csx

```diff
@@ -4,39 +4,130 @@
 #load "../util/tools_util.csx"
 
 using System;
+using System.Diagnostics;
 using System.IO;
 using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
 using System.Xml;
 
 // TODO: Use actual command line argument parser so we can have help text, etc...
-var branch = Args.Count() == 2 ? Args[0] : "master";
+var sourceFolder = Args[0];
 var destinationFolder = Args.Count() == 2 ? Args[1] : @"C:\Roslyn\Binaries\Release";
 
-var sourceFolder = $@"\\cpvsbuild\drops\Roslyn\Roslyn-{branch}-Signed-Release";
+Console.WriteLine("Kill the process or close the window to stop the Perf Test Automation");
 
-string latestBuild = null;
-foreach (var folder in Directory.GetDirectories(sourceFolder, "????????.?").Reverse())
+while (true)
 {
-    if (SanityTestPassesForBuild(folder))
+    try
     {
-        latestBuild = folder;
-        break;
+        // Get the last build that was tested
+        string lastBuild = null;
+        var perfRunStatusPath = Path.Combine((Environment.ExpandEnvironmentVariables("%USERPROFILE%")), "PerfRunStatus.txt");
+        if (File.Exists(perfRunStatusPath))
+        {
+            lastBuild = File.ReadAllText(perfRunStatusPath);
+        }
+
+        StartWatching(sourceFolder, destinationFolder, lastBuild, perfRunStatusPath);
+
+        // Wait for 10 minutes before looking for new build again
+        Console.WriteLine($"Waiting for 10 minutes from {DateTime.Now} to look for new builds");
+        Thread.Sleep(10 * 60 * 1000);
+        Console.WriteLine($"Resume looking for new builds in {sourceFolder}");
+    }
+    // Dont let any exception to stop our Perf Automation
+    catch (System.Exception e)
+    {
+        var logFilePath = Path.Combine((Environment.ExpandEnvironmentVariables("%USERPROFILE%")), "PerfRunErrorLog.txt");
+        File.AppendAllLines(logFilePath, new[] { DateTime.Now.ToString()});
+        File.AppendAllLines(logFilePath, new[] { e.Message});
+        File.AppendAllLines(logFilePath, new[] { e.StackTrace});
+
+        CleanupSystem();
     }
 }
 
-if (latestBuild == null)
+void CleanupSystem()
 {
-    throw new InvalidOperationException($"Could not locate build with passing tests at location \"{sourceFolder}\".");
-}
+    var processNamesToKill = new[] {"csc", "vbc", "vbcscompiler", "devenv", "cpc" };
+    System.Diagnostics.Process.GetProcesses()
+        .Where(x => processNamesToKill.Any(p => x.ProcessName.ToLower().StartsWith(p)))
+        .ToList()
```

Comment:
There is never-ending discussion to have a `Do`, which is pretty similar to `ForEach`.